index.js

```diff
@@ -39,30 +40,100 @@ function getTorrentMetaBuffer () { // TODO Fix parsing bug so this can be remove
 }
 
 function getFile (fpath) {
-  return _getFile(this, fpath)
+  return _getFile(module.exports, fpath)
+  .then(file => {
+    debug('FILE path=' + (file || {}).path, 'found=' + (file != null), file)
+    return file
+  })
+}
+
+function fetch (req, opts) {
+  opts = opts || {}
+  let inject = false
+  let url = null
+
+  // Convert a FetchEvent to Request
+  if (global.FetchEvent && req instanceof global.FetchEvent) {
+    inject = req.clientId == null
+    req = req.request // req is now an instance of Request
+  }
+
+  // Convert a Request to an URL
+  if (req instanceof global.Request) {
+    url = new URL(req.url)
+    inject = inject && url.search.substr(1)
+             .split('&').find(s => s === 'noPlanktosInjection') == null
+    if (req.method !== 'GET') throw new Error('Only HTTP GET requests supported')
+  } else if (req instanceof URL) {
+    url = req
+  } else if (typeof req === 'string') {
+    url = new URL(req)
+  }
+
+  if (req == null) throw new Error('Must provide a FetchEvent, Request, URL, or a string')
+  if (url.origin !== global.location.origin) throw new Error('Cannot Fetch. Origin differs')
+
+  inject = 'inject' in opts ? opts.inject : inject
+
+  debug('FETCH-REQ url=' + url.pathname, 'inject=' + inject)
+
+  // Generate response blob. Depends on if the downloader should be injected or not
+  let blobPromise = null
+  if (inject) {
+    let fname = url.pathname.substr(url.pathname.lastIndexOf('/') + 1)
+    const isHTML = fname.endsWith('.html') || fname.endsWith('.htm') || !fname.includes('.')
```

Comment:
I agree, but for a lack of a better way. The whole injector idea needs to be rethought to fix this.

The case that makes the problem difficult. The file is not in planktos so it will be fetched over http through the injector. But to know whether the requested file is actually html or not requires reading the 'Content-Type' header which requires a http fetch. So we can't pick the right injector until after we fetched the file, but it's supposed to be the injectors job to fetch the file.